### DISCUSSION TITLE
Instant track preview

### DISCUSSION THREAD

**Discussion by legsmechanical** [SynthstromAudible/DelugeFirmware#359] [Ideas]
**Proposal**:
When the sequencer is stopped, holding SHIFT and pressing and holding on a clip row in SONG view will immediately play back the respective clip (and no others) for as long as the row is held down.

This would allow you to quickly find the clip you want to work on without doing what I often find myself resorting to (especially when I've got tons of clips and sections): pressing play on the sequencer and soloing each clip until I find the one I'm looking for.  The Yamaha RS7000 does this, and it's a major boon to workflow.

**Comment by weavermedia** [comment]
This would be super handy as a feature but I also think it's just a workaround for the bigger issue of not being able to easily identify clips.

**Reply by legsmechanical** [reply]
I think that “bigger issue” is just kinda inherent to the machine, so not sure what the alternative would be. Naming, perhaps - but we already have that for instruments and that’s find if you’re willing to take the time on oled and not nearly as helpful if you’re doing a lot of cloning or on a 7seg.

**Reply by weavermedia** [reply]
I'm sure there are some improvements that could make organizing and identifying clips easier.

How about having a user defined list of clip types and the ability to assign each clip a type. The types could have assigned colors, so all BASS clips might be blue, all CHORD clips might be green, etc. That would make it easier to identify clips.

The ability to sort the Clip view by section and by clip type might be useful. Giving the user the ability to group all clips by section allows you yo work on one section at a time with the clips next to each other. Then sorting by type would put all the blue bass clips next to each other allowing you to quickly work on each bass part.

Neither of these ideas involves the display or naming the clips but, to me at least, sound like they would help a lot in my workflow. And the instant preview idea would be an added bonus too.

**Reply by legsmechanical** [reply]
Agree that'd all be great.  I'm just not sure track preview is a "workaround" in the sense that it provides a distinct workflow benefit apart from general clip organization.

**Reply by weavermedia** [reply]
Maybe workaround was the wrong word. I don't mean to dismiss the idea. It's sounds like a really useful workflow addition and I'm pretty sure I'd use it a lot.

I agree that this feature stands on its own without being sold as a solution to clip organization.
